pkg/puller/downloader.go

```diff
@@ -0,0 +1,28 @@
+package puller
+
+import (
+	"fmt"
+	"gopkg.in/matryer/try.v1"
+	"log"
+	"time"
+)
+
+func DownloadModel(numRetries int, event EventWrapper) error {
+	err := try.Do(func(attempt int) (bool, error) {
+		err := download(event.ModelDef.StorageUri)
+		if err != nil {
+			time.Sleep(1 * time.Second) // wait a second
```

Comment:
Why do we wait for a second here?

Reply:
Just a dummy thing for retry purposes, I don’t mind if we retry immediately. It’s a WIP